### ISSUE TITLE
[Reference] Missing sections

### ISSUE THREAD

**Issue by franramirez688** [conan-io/docs#2943]
* [ ] Configuration files (global.conf, conandata.yaml, profiles, etc) (@franramirez688 WIP)
* [ ] Environment variables (CLICOLOR_FORCE, CONAN_HOME, etc)
* [x] Hooks (Now, it's in `Reference -> Extensions -> Hooks`)
* [ ] Commands
     * [ ] conan config home
     * [ ] conan graph (everything)
     * [ ] conan inspect
     * [ ] conan lock
     * [ ] conan list (HTML format - depends on Conan development WIP)
     * [ ] conan profile
     * [ ] conan remote
     * [ ] conan search (review it!)
     * [ ] conan create
     * [ ] conan export
     * [ ] conan test
     * [ ] conan upload
     * [ ] conan build
* [ ] Extensions
      * [ ] Python requires
      * [ ] Python API (almost all the sub-sections)
      * [x] Binary compatibility (it's there, but should we add a more complex example?)
      * [ ] Profile plugin (we only have to add the `profile` link to its reference)
* [ ] Recipe Tools
      * [ ] conan.tools.files.rmdir() (only docstring in Conan)
      * [ ] conan.tools.files downloads (add link to `global.conf`)
      * [ ] MesonDeps (docstrings in Conan)
      * [ ] MSBuildToolchain (docstrings in Conan)
      * [ ] NMakeXXXXX (review all their docstrings)
      * [ ] vs_layout (poorly explained)
      * [ ] conan.tools.microsoft.visual.unix_path (docstrings in Conan)
      * [ ] conan.tools.microsoft.visual.subsytems (totally missing)
      * [ ] conan.tools.scm.Version (review docstrings)

Feel free to add/edit missing/wrong parts. Furthermore, feel free to put your own name (for instance "@xxxxxx WIP") for each checkbox if you're working on a new PR for that section to avoid collisions

**Comment by Croydon** [comment]
> Python API (almost all the sub-sections) <- Dont document them for now

The "don't document them now" part is now outdated after 2.0 is released? Looking forward to this part in particular ✌️

**Comment by SpaceIm** [comment]
Yep, I would love to see documentation of Python API. I have no idea how to use those things, and what I'm manipulating and what can be done.
